gobblin-hive-registration/src/main/java/org/apache/gobblin/hive/orc/HiveOrcSerDeManager.java

```diff
@@ -264,7 +272,18 @@ private void addSchemaProperties(Path path, HiveRegistrationUnit hiveUnit)
    *
    */
   protected void addSchemaPropertiesHelper(Path path, HiveRegistrationUnit hiveUnit) throws IOException {
-    TypeInfo schema = getSchemaFromLatestFile(path, this.fs);
+    TypeInfo schema;
+    if(props.getPropAsBoolean(HIVE_SPEC_SCHEMA_FROM_WRITER, DEFAULT_HIVE_SPEC_SCHEMA_FROM_WRITER)) {
+      try {
+        Preconditions.checkArgument(props.contains(WRITER_LATEST_SCHEMA));
+        Schema avroSchema = new Schema.Parser().parse(props.getProp(WRITER_LATEST_SCHEMA));
+        schema = TypeInfoUtils.getTypeInfoFromObjectInspector(new AvroObjectInspectorGenerator(avroSchema).getObjectInspector());
```

Comment:
There is a Avro->Orc schema converter utility method in AvroOrcSchemaConverter class. Can we use that, given that it is being used by GobblinOrcWriter to create the ORC file?

Reply:
The schema converter returns a TypeDescription object. Not sure if it is easy to go from TypeDescription to TypeInfo.

Reply:
Yeah I was trying to do that. Several reasons here: 
1. AvroOrcSchemaConverter is now defined in gobblin-orc module, I don't think it make sense for us to introduce new dependency for hive registration module.
2.  It's doable to transfer TypeDescription to TypeInfo, but it's the same way that we need to use OrcUtils to create one objectInspector and get typeInfo there. As we are using the writer schema to get orcSchema, I think the two results should be the same? I verified one table and it looks good to me.
What do you think?

Reply:
Makes sense. Thanks for the explanation!